server/graphql/v2/interface/Account.js

```diff
@@ -393,6 +393,29 @@ const accountFieldsDefinition = () => ({
   location: {
     type: Location,
     description: 'The address associated to this account. This field is always public for collectives and events.',
+    async resolve(collective, _, req) {
+      const canSeeLocation =
+        collective.type !== CollectiveTypes.USER ||
+        (req.remoteUser?.isAdmin(collective.id) && checkScope(req, 'account')) ||
+        (await collective.isHost());
```

Comment:
These permissions are moved and adapted from Individual.js. The `isHost()` call seemed unintuitive to me, but looks like it makes sure to expose location for USER's that are also hosts, will check if this actually happens..

Worth noting is that there currently is a discrepancy between GQL v1 and v2 for what Collective types they expose location, v1 exposes for `COLLECTIVE`, `EVENT`, `ORGANIZATION`, and v2 for all types except `USERS`.

Reply:
> I wonder why we moved the location resolver to this file rather than keeping it in Individual.location; it's now unclear where one should look at for the resolver code. Like, if I'm returning an organization, will this be used or Organization.location's resolver?

It was moved since all Account types now need a call to `getLocation()` or rather the newly added loader.

Good point about ambiguous resolving for `Organization.location`, I overlooked also removing that one, which I've now done, so all logic/permissions regarding location resolving is now in the Account interface.

Simply removing the `Organization.location` resolver results in a small change for permissions. Only the country was exposed in v2 if you were not a host. However, v1 exposed the full location for all Organizations, so we're not exposing any previously private data if we choose to let location be public for all Account types except for Users in v2 (which seems like the simpler/better approach)